### DISCUSSION TITLE
Having issues when trying to upload an xml to presigned url - SP-API

### DISCUSSION THREAD

**Discussion by bodamithun** [amzn/selling-partner-api-models#2813] [General]
Hi, I am working on feeds. I successfully finished calling createfeeddocument  and in response I got the feeddocumentid and url, to which I need to upload the feed document. so when I am trying to upload and xml, I get a 403 access denied. I checked my developer profile and I have all the permissions checked. Not sure what I am missing, can some body please guide as to what I am missing.

I would really appreciate.

Thanks

**Comment by albinsopaj** [comment]
Hi, there. Did you manage to solve your problem. I just recently ran into it, and I really don't know what to do. Thanks in advance.

**Comment by stefnats** [comment]
Could you share (anonymized) request data, like what do you send, what headers and also whats the exact response?

Seems more like you run into an issue with S3, not SP-API.
Basically, you just have to send a `PUT` request to the URL that you get from the `CreateFeedDocument` with the Contents from your file as request body.

**Reply by wszwc** [reply]
Hello, I also encountered this problem when I used feedApi to submit json product data. This is my request data, although I think this has nothing to do with the data itself. . . Please help me, thank you
[Uploading content.json…]()

**Comment by wszwc** [comment]
Hello, I also encountered the same problem. I used the json format. Is there any solution?
